test_data/yast/yast2_gui.yaml

```diff
@@ -0,0 +1,30 @@
+bootcode_options:
+  bootloader_type: GRUB2
+  write_to_partition: 0
+  write_to_mbr: 1
+  select_custom_boot_partition: 0
+  set_active_flag: 1
+  write_generic_to_mbr: 0
+  trusted_boot_support: 0
+  protective_mbr_flag: "do not change"
+applied_params:
```

Comment:
I just want to share here my view, you don't have to change anything at all :). I'm just analyzing what you did once I asked you to put in test data each configuration.

So the general idea is to have test data that when you read them easily we will see how is configured this scenario. It is clear here than the test data is repeated, for one side we have test data for the UI and the other side we have test data for the console validation (we have this problem in general in other tasks, nothing particular of this one). How this could be avoided? under 'bootcode_options' we could just include the stages that we need, like you did with 'verify_current_options', 'write_generic_to_mbr' and under those we just write the keys that you have under bootloader_options. And that would be all from test data perspective, simple and readable. Then we need a library or a function (probably not in the test itself) which would translate those test data to the structure you have written with path and entries. So yes, putting back in code as you had it before but in a general way, creating a function that is able to map whatever is supported. Also `grub_device` seems to be out of the 'bootcode_options'. 

I will share similar approach I did in other PR and for the future I believe we should start by defining test data before writing any code if possible.

Despite of that I liked how well organized the test data is now and you did a good job indeed, I'm just thinking forward how things would scale better.

Reply:
Thanks ! I'm not sure to understand fully what you suggest, it sounds like probably good idea but also like that would be again some pretty big refactoring at that point, so I'll keep as-is for now :)